projects/wave-core/src/lib/layers/layer-icons/line-icon/line-icon.component.ts

```diff
@@ -0,0 +1,45 @@
+import {Component, OnInit, ChangeDetectionStrategy, Input, SimpleChanges, OnChanges} from '@angular/core';
+import {StrokeDashStyle} from '../../symbology/symbology.model';
+import {BLACK, Color} from '../../../colors/color';
+
+export interface LineIconStyle {
+    strokeWidth: number;
+    strokeDashStyle: StrokeDashStyle;
+    strokeRGBA: Color;
+}
+
+@Component({
+    selector: 'wave-line-icon',
+    templateUrl: './line-icon.component.svg',
+    styleUrls: ['./line-icon.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LineIconComponent implements OnInit, OnChanges {
+
+    @Input()
+    iconStyle: LineIconStyle;
+    strokeWidth = 2;
+    strokeDashArray: Array<number> = [];
+    strokeColor = BLACK.rgbaCssString();
+
+    constructor() {
+    }
+
+    ngOnInit(): void {
+        if (this.iconStyle) {
```

Comment:
müsste das nicht `ngOnUpdate` sein? Also falls sich der Input mal ändert.

Reply:
Ich sehe, beim PointIcon hast du das auch gemacht.

Reply:
und bei PolygonIcon machst du beides, also es sollte konsistent sein ^^.

Reply:
all icon components handle onInit and OnChanges...